react-components/src/communities/view/CommunityListing.js

```diff
@@ -44,10 +45,19 @@ class CommunityListing extends Component {
         this.setState({order, orderBy});
     }
 
+    handleRowClick(community) {
+        this.setState({
+            selectedCommunity: community
+        });
+
+        this.props.onCommunitySelected(community);
```

Comment:
Since we're passing the selected community back up to the `ManageCommunitiesView`, we could also have that view pass the selected community back down to this component as a prop, rather than also storing it in this component's state.

I'm not sure if there's a benefit of storing the selected community in this component's state vs. getting it from props, but I get the impression that React prefers controlling components with props over state (where it makes sense). In this case, it could also open up the possibility of telling this listing component that a community has been selected in some way other than the user clicking a row (though I don't know if we'll ever need that functionality 🤔 ).

Reply:
Sure, I'll refactor!  Did we still want a design discussion on our state management strategy going forward?  With communities, I'm definitely managing most of it in the React components instead of in GWT, similar to what we had in the Analysis window, so I may need more refactors depending on what we decide.

Reply:
We can. I'm not sure we can nail down an overarching strategy (beyond what I just mentioned above with props over state where reasonable), but if we have questions or want further clarifications about it, let's discuss it!